core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -33,294 +34,228 @@
+ * The class is responsible for the actual sending of
+ * {@link com.microsoft.applicationinsights.internal.channel.common.Transmission}
  *
  * The class uses Apache's HttpClient framework for that.
  *
  * Created by gupele on 12/18/2014.
  */
 public final class TransmissionNetworkOutput implements TransmissionOutput {
-    private final static String CONTENT_TYPE_HEADER = "Content-Type";
-    private final static String CONTENT_ENCODING_HEADER = "Content-Encoding";
-    private final static String RESPONSE_THROTTLING_HEADER = "Retry-After";
-    private final static String RESPONSE_RETRY_AFTER_DATE_FORMAT = "E, dd MMM yyyy HH:mm:ss";
+	private final static String CONTENT_TYPE_HEADER = "Content-Type";
+	private final static String CONTENT_ENCODING_HEADER = "Content-Encoding";
+	private final static String RESPONSE_THROTTLING_HEADER = "Retry-After";
+
+	private final static String DEFAULT_SERVER_URI = "https://dc.services.visualstudio.com/v2/track";
+
+	// For future use: re-send a failed transmission back to the dispatcher
+	private TransmissionDispatcher transmissionDispatcher;
+
+	private final String serverUri;
+
+	private volatile boolean stopped;
+
+	// Use one instance for optimization
+	private final ApacheSender httpClient;
+
+	private TransmissionPolicyManager transmissionPolicyManager;
+
+	/**
+	 * Creates an instance of the network transmission class.
+	 * <p>
+	 * Will use the DEFAULT_SERVER_URI for the endpoint.
+	 * @param transmissionPolicyManager The transmission policy used to mark this sender active or blocked.  
+	 * @return
+	 */
+	public static TransmissionNetworkOutput create(TransmissionPolicyManager transmissionPolicyManager) {
+		return create(DEFAULT_SERVER_URI, transmissionPolicyManager);
+	}
+
+	/**
+	 * Creates an instance of the network transmission class.
+	 * 
+	 * @param endpoint The HTTP endpoint to send our telemetry too.
+	 * @param transmissionPolicyManager The transmission policy used to mark this sender active or blocked.  
+	 * @return
+	 */
+	public static TransmissionNetworkOutput create(String endpoint,
+			TransmissionPolicyManager transmissionPolicyManager) {
+		String realEndpoint = Strings.isNullOrEmpty(endpoint) ? DEFAULT_SERVER_URI : endpoint;
+		return new TransmissionNetworkOutput(realEndpoint, transmissionPolicyManager);
+	}
+
+	/**
+	 * Private Ctor to initialize class.
+	 * <p>
+	 * Also creates the httpClient using the ApacheSender instance
+	 * @param serverUri The HTTP endpoint to send our telemetry too.
+	 * @param transmissionPolicyManager
+	 */
+	private TransmissionNetworkOutput(String serverUri, TransmissionPolicyManager transmissionPolicyManager) {
+		Preconditions.checkNotNull(serverUri, "serverUri should be a valid non-null value");
+		Preconditions.checkArgument(!Strings.isNullOrEmpty(serverUri), "serverUri should be a valid non-null value");
+		Preconditions.checkNotNull(transmissionPolicyManager,
```

Comment:
What are scenarios where this could be null. How does code flow happen in case of this event?

Reply:
The Preconditions statements are for us. Callers of this method should make sure `serverUri` is not null before calling.